### ISSUE TITLE
Recreating the 6 seconds for 10M edges runtime

### ISSUE THREAD

**Issue by ariellubonja** [cshen6/GraphEmd#7]
Hi!

A runtime on the [LiveJournal graph](https://snap.stanford.edu/data/soc-LiveJournal1.html) with 70M edges took 4:58 minutes on our lab computers using the Python GEE version

I experienced a 5m+ runtime on Pubmed on Matlab, and that graph has only 44,000 edges. I am less familiar with Matlab so I may be doing something wrong

I ran it using: `inputPubmed = load("pubmedAdj.mat")` and `pubmed=GraphEncoderEvaluate(inputPubmed.Adj,inputPubmed.Y);`  Memory usage spiked above 10GB

I presumed Matlab was the most optimized based on you saying most experiments were run on Matlab. Didn't run any R code

Am I doing something wrong? How can I recreate the 6 seconds for 10 million edges quoted in the paper?

Thank you!

**Comment by cshen6** [comment]
Hi Ariel,

GraphEncoderEvaluate actually evaluate the classification error on 10 fold
cross validation.
And it has a number of methods there, not just GEE...(though you can turn
off spectral method in the option).

Running it one time is: GraphEncoder(inputPubmed.Adj,inputPubmed.Y), which
should be a few seconds.

**Comment by ariellubonja** [comment]
Thank you! Seems to run much faster on the PubMed dataset now. Will download a 10M graph from network repository and update!

**Comment by ariellubonja** [comment]
Hi! I finally got access to a 24-core, 32GB machine. I'm getting a 28-second runtime on the [LiveJournal](http://snap.stanford.edu/data/soc-LiveJournal1.html) graph, which is a linear scaling from the 10M for 6seconds claim!

1. In case it's helpful, what I did, beside switching machines, is I pulled your new code and installed the [Matlab Parallel Toolbox](https://www.mathworks.com/products/parallel-computing.html) package (not sure if it helped)

Thank you!
